### ISSUE TITLE
Assigning the value to an intermediate signal returns error

### ISSUE THREAD

**Issue by curryrasul** [namnc/circom-2-arithc#78]
I tried the circuit:
```circom
pragma circom 2.1.0;

template temp() {
    signal input a;
    signal output out;

    signal intermediate <== 0;
}

component main = temp();
```

It finishes with error:
```bash
thread 'main' panicked at src/compiler.rs:520:51:
no entry found for key
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by namnc** [comment]
I think I fixed this with https://github.com/namnc/circom-2-arithc/commit/6e280217b1662350fd550f55ffb42b89fbfe3bda. BUT up to @curryrasul to confirm.

The issue should be due to 2 things:
- The naming of a constant signal does not take into account multiple instances of a template, e.g. Switcher.const_signal_0 appears multiple times with different signal ids --> naming could be more robust with differentiation between instances, e.g. Switcher_0.const_signal_0 and Switcher_1.const_signal_1
- This leads to merging based on constant signal names failed as now every signal has the same name are merged into one even though it should be all of them --> merging can be more robust, e.g. by appending signal id at the end we can keep every const signal, e.g. Switcher.const_signal_0_18, Switcher.const_signal_0_24. --> ALSO what I have done inside the fix.

**Comment by namnc** [comment]
I think this is a weird circuit may be we give it low priority ...

**Comment by curryrasul** [comment]
@namnc Yes, agree. It's not only in this circuit though, it's just an example
